crates/engine/tree/src/tree/invalid_block_hook.rs

```diff
@@ -49,3 +49,31 @@ impl InvalidBlockHook for NoopInvalidBlockHook {
     ) {
     }
 }
+
+/// A chain of [`InvalidBlockHook`]s that are executed in order.
+pub struct InvalidBlockHookChain(pub Vec<Box<dyn InvalidBlockHook>>);
```

Comment:
can we replace `Chain` with something else, maybe even just `InvalidBlockHooks`

Reply:
no chain in blockchain project

![image](https://github.com/user-attachments/assets/1e3d7d5e-dcdd-4c9d-a42e-e884cad89c46)

Reply:
but agreed, just `InvalidBlockHooks` is easier